framework/include/interfaces/TaggingInterface.h

```diff
@@ -132,6 +132,9 @@ class TaggingInterface
 
 protected:
   /// The vectors this Kernel will contribute to
+  std::set<TagID> _vector_tag_ids;
```

Comment:
See above. We basically need to be careful on the difference on `_id` and `_type_id` of `VectorTag`. Previously the two do not have difference because `_type_id` is a subset and always match `_id`. Remember I mentioned on slack that there could be a bug somewhere in the code not differentiating the two properly, which I have really hard time to figure out, after I fix something obvious in `TaggingInterface`. My current design does not trigger that, so I will leave that as part of our future work of maintaining tagged solutions and residuals separately.